### PR TITLE
Use ECR as go lib instead of the binary

### PR DESCRIPTION
**Pull request by Rowern**
It might be possible to do the same with GCR but for now I don't have a google cloud account, I'm might be able to create one for this purpose but for now I tested ECR.

### CODE REVIEW

**Comment by Rowern** [comment]
I have no way of testing GCR (even after doing the whole setup and loosing 2h with service accounts and their horrible IAM console...).
If anyone is willing, could you just try this PR to validate that GCR is still working properly ?

File: lib/registry/security/security.go [hunk 1]
```diff
@@ -120,6 +126,34 @@ func (c Config) getCredentials(helper, addr string) (types.AuthConfig, error) {
 }
 
 func (c Config) getCredentialFromHelper(helper, addr string) (types.AuthConfig, error) {
+	if strings.HasPrefix(helper, "ecr") {
```
Comment (evelynl94) [inline lib/registry/security/security.go]: maybe match the full name `ecr-login` would be better here?

Comment (Rowern) [reply lib/registry/security/security.go]: Switched to switch case

**Comment by Rowern** [comment]
Sorry for the `push -f`, now this is updated with latest master (so no more GCR auth bug)

**Comment by evelynl94** [comment]
thank you for doing that!

**Pull request merged**